### PR TITLE
Add new region Instance Template data source

### PR DESCRIPTION
**Pull request by jingcongw**
Background
Regional Instance Template (later: RIT) is a new regional resource, implementing the functionality of [Instance Templates](https://cloud.google.com/compute/docs/reference/rest/v1/instanceTemplates) existing before only in the global scope. More details can be found: [go/rit-support-in-terraform](https://goto.google.com/rit-support-in-terraform).

Changes include:
1. Add Region Instance Template datasource, acceptance tests and document.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_compute_region_instance_template`
```

### CODE REVIEW

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			fmt.Println(err)
+			return err
+		}
+
+		var items []*compute.InstanceTemplate
+
+		if err := json.Unmarshal(instanceTemplatesList, &items); err != nil {
+			fmt.Println(err)
+			return err
+		}
```
Comment (zli82016) [inline mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb:97]: Do we need this block?

File: mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+* `id` - an identifier for the resource with format `projects/{{project}}/regions/{{region}}/instanceTemplates/{{name}}`
+
+* `metadata_fingerprint` - The unique fingerprint of the metadata.
+
+* `self_link` - The URI of the created resource.
+
+* `tags_fingerprint` - The unique fingerprint of the tags.
+
+[1]: /docs/providers/google/r/compute_instance_group_manager.html
+[2]: /docs/configuration/resources.html#lifecycle
```
Comment (zli82016) [inline mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown:305]: What is the reason to add the docs here? I think we can delete them.

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb [hunk 3]
```diff
@@ -0,0 +1,119 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"encoding/json"
```
[the remark this answers is not in the document]
Comment (jingcongw) [reply mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb:5]: They are one tab indentation and I have ran make lint without issue.

File: mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown (hunk 2 above)
Comment (jingcongw) [reply mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown:305]: It's the user guide for the new data source. Do you mean it's the wrong place to place this doc?

File: mmv1/third_party/terraform/tests/data_source_google_compute_region_instance_template_test.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
+package google
+
+import (
+	"testing"
```
[the remark this answers is not in the document]
Comment (jingcongw) [reply mmv1/third_party/terraform/tests/data_source_google_compute_region_instance_template_test.go:4]: Yes. I have ran make lint to verify.

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+			return fmt.Errorf("error retrieving list of region instance templates: %s", err)
+		}
+
+		instanceTemplates := templates["items"]
+
+		instanceTemplatesList, err := json.Marshal(instanceTemplates)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
```
[the remark this answers is not in the document]
Comment (jingcongw) [reply mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb:90]: Yes. This block is needed to encode response data in json and convert to the desired type then stores in the items collection.

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb (hunk 1 above)
Comment (jingcongw) [reply mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb:97]: Answered above. Thanks!

File: mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown:305]: The two docs are not needed, as the user does not need to know the code source.

Comment (jingcongw) [reply mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown]: Oh, I see. Corrected. Thanks for catching it!

**Comment by jingcongw** [comment]
@zli82016 Could you help run /gcbrun again? I pushed some new changes to clean up the documentation. Thanks!

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**